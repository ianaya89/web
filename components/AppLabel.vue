<template>
  <v-chip
    :class="[{hidden: isClosed, 'cursor-pointer': closeable}]"
    :close="closeable"
    v-bind="$props"
    class="app-label"
    @input="toggleClose"
  >
    <slot>Label</slot>
    <span v-if="closeable" class="label-close" @click.stop="toggleClose">
      <span class="pl-3 pt-1 -mr-1 lni-close" />
    </span>
  </v-chip>
</template>

<script>
export default {
  name: 'AppLabel',

  props: {
    closeable: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean,
      default: false
    },
    outline: {
      type: Boolean,
      default: false
    },
    small: {
      type: Boolean,
      default: false
    },
    color: {
      type: String,
      default: null
    },
    textColor: {
      type: String,
      default: null
    }
  },

  data: () => ({
    isClosed: false
  }),

  methods: {
    toggleClose() {
      if (!this.disabled) {
        this.isClosed = !this.isClosed
        if (this.isClosed) {
          this.$emit('close')
        }
      }
    }
  }
}
</script>

<style lang="postcss">
.v-chip.app-label {
  @apply .rounded-sm .text-white .text-p3 .bg-black .tracking-wide .font-medium .pt-1 .h-8 .uppercase;
  transition: all 0.3s;

  &:hover,
  &:focus,
  &:active {
    @apply .shadow .bg-grey;
  }

  .v-chip__content {
  }

  &.label-blue {
    @apply .text-black .bg-blue-l5;
  }

  &.label-green {
    @apply .text-black .bg-green-l5;
  }

  &.label-red {
    @apply .text-black .bg-red-l5;
  }

  &.label-yellow {
    @apply .text-black .bg-yellow;
  }
}

.v-chip.v-chip.v-chip--outline {
  @apply text-black;

  &:hover,
  &:focus,
  &:active {
    border: 1px solid grey;
    color: grey;
    box-shadow: none;
  }
}

.v-chip.app-label.v-chip--disabled {
  @apply .bg-grey-l2;
}

.v-chip__close {
  i {
    display: none;
  }
}

.label-close {
  @apply .text-red;
  transition: 0.3s ease-out;
}
</style>
