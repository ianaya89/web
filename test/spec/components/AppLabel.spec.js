import { mount, shallowMount } from '@vue/test-utils'
import AppLabel from '@/components/AppLabel'

describe('AppLabel', () => {
  it('renders default label', () => {
    const wrapper = shallowMount(AppLabel)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders some props', () => {
    const wrapper = shallowMount(AppLabel, {
      propsData: {
        label: 'Custom label',
        textColor: 'white',
        color: 'secondary',
        disabled: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders with closing', () => {
    const wrapper = mount(AppLabel, {
      propsData: {
        closeable: true,
        disabled: true
      }
    })

    expect(wrapper.element).toMatchSnapshot()

    wrapper.find('.label-close').trigger('click')
    expect(wrapper.emitted('close')).toBeUndefined()
    expect(wrapper.element).toMatchSnapshot()

    wrapper.setProps({ closeable: true, disabled: false })
    wrapper.find('.label-close').trigger('click')
    expect(wrapper.emitted('close')).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})
