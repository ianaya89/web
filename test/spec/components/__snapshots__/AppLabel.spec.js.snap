// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLabel renders default label 1`] = `
<v-chip-stub
  class="app-label"
  value="true"
>
  Label 
  <!---->
</v-chip-stub>
`;

exports[`AppLabel renders some props 1`] = `
<v-chip-stub
  class="app-label"
  color="secondary"
  disabled="true"
  label="Custom label"
  textcolor="white"
  value="true"
>
  Label 
  <!---->
</v-chip-stub>
`;

exports[`AppLabel renders with closing 1`] = `
<span
  class="v-chip app-label v-chip--disabled v-chip--removable theme--light cursor-pointer"
  closeable="true"
  tabindex="-1"
>
  <span
    class="v-chip__content"
  >
    Label 
    <span
      class="label-close"
    >
      <span
        class="pl-3 pt-1 -mr-1 lni-close"
      />
    </span>
    <div
      class="v-chip__close"
    >
      <i
        aria-hidden="true"
        class="v-icon material-icons theme--light"
      >
        cancel
      </i>
    </div>
  </span>
</span>
`;

exports[`AppLabel renders with closing 2`] = `
<span
  class="v-chip app-label v-chip--disabled v-chip--removable theme--light cursor-pointer"
  closeable="true"
  tabindex="-1"
>
  <span
    class="v-chip__content"
  >
    Label 
    <span
      class="label-close"
    >
      <span
        class="pl-3 pt-1 -mr-1 lni-close"
      />
    </span>
    <div
      class="v-chip__close"
    >
      <i
        aria-hidden="true"
        class="v-icon material-icons theme--light"
      >
        cancel
      </i>
    </div>
  </span>
</span>
`;

exports[`AppLabel renders with closing 3`] = `
<span
  class="v-chip app-label v-chip--disabled v-chip--removable theme--light cursor-pointer"
  closeable="true"
  tabindex="-1"
>
  <span
    class="v-chip__content"
  >
    Label 
    <span
      class="label-close"
    >
      <span
        class="pl-3 pt-1 -mr-1 lni-close"
      />
    </span>
    <div
      class="v-chip__close"
    >
      <i
        aria-hidden="true"
        class="v-icon material-icons theme--light"
      >
        cancel
      </i>
    </div>
  </span>
</span>
`;
